corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js

```diff
@@ -727,6 +727,7 @@ hqDefine('app_manager/js/forms/case_config_ui', function () {
                 return usercaseTransaction({
                     case_properties: case_properties,
                     case_preload: case_preload,
+                    case_type: 'commcare-user', // will get overridden by the default. Set here to check deprecated status for saved properties on initial page load
```

Comment:
@zandre-eng 
Can you tell more about why this change was made?
Is usercaseTransaction case_type actualy 'commcare-user'?

Reply:
Unlike the properties being saved under "Case Management" which go to whichever case type was selected in the case list, the properties from "User Properties" only get saved to the `commcare-user` case type. Since it's a constant, I thought it would be safe to put here as well. 

The reason we need it here is that we need `case_type` to be pre-defined so that we can do the deprecation check on the initial page load (which happens before the `usercaseTransaction` model has a chance to define its own `case_type` property.

Reply:
Can we do an alternate approach here?
I am not sure about populating this field with a constant so as to satisfy a validation check.
Is the validation check not relevant for `usercaseTransaction`? Can we skip that check for it then since we know it would not have a value set for case_type.

Reply:
> Can we skip that check for it then since we know it would not have a value set for case_type.

Do you mean skip showing a deprecated warning for case properties in the "User Properties" tab? It would be nice if we can still do the check for user properties. Users are able to deprecate user properties in the DD, so there could potentially be deprecated case properties that are selected. It would keep the UI consistent with the "Case Management" tab.

We do use the constant in other places as well, such as [here](https://github.com/dimagi/commcare-hq/blob/b23df94313306caf3c81826993938fa96dedb35f/corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js#L487) and [here](https://github.com/dimagi/commcare-hq/blob/b23df94313306caf3c81826993938fa96dedb35f/corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js#L721). Since this constant string is already being used I wonder then if it might be sufficient to have a global constant `COMMCARE_USER_CASE_TYPE` and then simply refer this throughout the file? That way, if it ever changes we only have to do it in one place in the file.

Reply:
is commcare-user the expected case type for usercaseTransaction but just set [later](https://github.com/dimagi/commcare-hq/blob/b23df94313306caf3c81826993938fa96dedb35f/corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js#L487) instead?

Reply:
That is correct. For case properties, the case type gets set dynamically [here](https://github.com/dimagi/commcare-hq/blob/b23df94313306caf3c81826993938fa96dedb35f/corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js#L462), whereas user properties have a hard-coded case type set where you've linked in your comment.

Reply:
okay, thank you :+1: